File: songs/python/12-unique-a.py
from MySongs import songs

tags = ['60s', 'jazz', '60s', 'rock',
        '70s', 'rock', '70s', 'pop']

def exclude(val, tags):
  return [tag for tag in tags if tag != val]

def unique(tags):
  if len(tags) <= 1: return tags
  else:
    head, *tail = tags
    return [head] + unique(exclude(head, tail))

print(unique(tags))

File: songs/perl/12-unique-a.pl
#!/usr/bin/perl
use 5.010;
use strict;
use warnings;

use File::Basename;
use lib dirname(__FILE__);

use MyHelperUnique;

my @tags = ('rock',
  'jazz', 'rock', 'pop', 'pop');

say join(':',
  MyHelperUnique::unique(@tags));
